includes/utilities.inc

```diff
@@ -634,6 +634,7 @@ function islandora_solr_is_date_field($solr_field) {
   $date_types = array(
     'org.apache.solr.schema.DateField',
     'org.apache.solr.schema.TrieDateField',
+    'org.apache.solr.schema.DatePointField',
```

Comment:
Since Solr 5.x there are also other changes in field types relevant to dates and ranges. One in particular for Multi valued dates i care for is
https://lucene.apache.org/solr/6_4_0/solr-core/org/apache/solr/schema/DateRangeField.html